pkgs/development/libraries/qt-5/5.15/default.nix

```diff
@@ -180,6 +180,17 @@ let
       ./qtwebengine-darwin-no-platform-check.patch
       ./qtwebengine-mac-dont-set-dsymutil-path.patch
       ./qtwebengine-darwin-checks.patch
+    ] ++ lib.optionals stdenv.cc.isClang [
```

Comment:
We prefer to make patches unconditional when possible.

Reply:
@emilazy What is the benefit of that? The current failure is very specific to clang and hiding the patch behind the conditional ensures that this pull request does not break on anything else and is easier to remove in the future in my opinion. Besides, I don't know if it will still build on gcc with linux since there harfbuzz seems to be dev-vendored: https://github.com/NixOS/nixpkgs/pull/358601#issuecomment-2495766115.

You are definitely more an expert when it comes to qt5 so I am fine to remove the conditional if you think that it is a better approach.

Reply:
With all that said, there is one main benefit to conditionals: they can avoid rebuilds so that things can go directly to `master`. In this case that could be beneficial as otherwise 24.11 will have a broken Qt 5 WebEngine for a few weeks after release. But frequently people will add conditionals to avoid rebuilds and then not get around to cleaning them up on `staging`, so it’s easier to just tell people to target `staging` directly and skip the conditionals :)

If you want to do the additional effort, then you could retarget this to `master`, keep the conditional, and then send a PR to `staging` after the periodic merge goes through to drop it. But usually things are not so time‐sensitive that they can’t wait a week or two to make the process simpler and avoid a pile‐up of conditionals.

Reply:
@emilazy Thank you for the detailed explanation👍 . Makes a lot of sense.